test/pylib/scylla_cluster.py

```diff
@@ -145,23 +145,21 @@ def __init__(self, exe: str, vardir: str,
         # Calculated in `install` as only then we know the seed servers.
         self.config: Dict[str, object] = {}
 
-        async def stop_server() -> None:
-            if self.is_running:
-                await self.stop()
-
-        async def uninstall_server() -> None:
-            await self.uninstall()
-
-        self.stop_artifact = stop_server
-        self.uninstall_artifact = uninstall_server
-
     async def install_and_start(self, api: ScyllaRESTAPIClient) -> None:
         """Setup and start this server"""
-        await self.install()
+        try:
+            await self.install()
+        except:
+            await self.uninstall()
```

Comment:
What is the idea of hiding the problem under the carpet if start fails? The results should be left available for investigation.

Reply:
You're commenting on `install`/`uninstall` pair. There's nothing to look at if `install` fails. And we don't `uninstall` if `start` fails: see the next `try` block. So if start fails, the results are available.